src/puya/awst/nodes.py

```diff
@@ -1488,6 +1797,12 @@ def accept(self, visitor: ExpressionVisitor[T]) -> T:
 
 @attrs.frozen
 class Reversed(Expression):
+    """
+    Returns an iterator-like expression which can be used in a ForInLoop. The iterator emits all
```

Comment:
Similar vibe to Range here, it's a lazy-sequence, and only usable with ForInLoop.